ucsmsdk/utils/comparesyncmo.py

```diff
@@ -0,0 +1,555 @@
+            continue
+
+        # not include operational property
+        if not include_operational:
+            if mo_property_meta.access in (MoPropertyMeta.INTERNAL,
+                                           MoPropertyMeta.READ_ONLY):
+                skip_props.append(prop)
+        # checks if property is part of current or earlier ucsm schema
+        if not no_version_filter:
+            version = mo.get_handle().version
+            if version is None or version < mo_property_meta.version or \
+               mo_property_meta.access == MoPropertyMeta.INTERNAL:
+                skip_props.append(prop)
+    return skip_props
+
+
+def _compare_known_mo(from_mo, to_mo, diff, include_operational=False,
+                      no_version_filter=False):
+    """
+    Internal function to compare if both the ref and diff obj is known mo.
+    """
+
+    from_mo_skip_props = None
+    if not include_operational or not no_version_filter:
+        from_mo_skip_props = _get_skip_props(from_mo, include_operational,
+                                             no_version_filter)
+
+    # comparing known properties of ref mo
+    for prop in from_mo.prop_meta:
+        if from_mo_skip_props and prop in from_mo_skip_props:
+            continue
+        # if not exist in diff mo
+        if not hasattr(to_mo, prop):
+            log.debug("Property '%s' of '%s' does not exist in diff obj." % (
+                prop, from_mo.dn))
+            continue
+
+        if getattr(from_mo, prop) != getattr(to_mo, prop):
+            diff.append(prop)
+
+    # comparing unknown properties of ref mo
+    from_xtra_props = from_mo._ManagedObject__xtra_props
+    to_xtra_props = to_mo._ManagedObject__xtra_props
+    if not to_xtra_props:
+        return
+    for prop in from_xtra_props:
+        if prop not in to_xtra_props:
+            continue
+
+        if from_xtra_props[prop].value != to_xtra_props[prop].value:
+            if not no_version_filter:
+                UcsWarning("Ignoring xtra property '%s' of '%s'" % (
+                    prop, from_mo.dn))
+            else:
+                diff.append(prop)
+
+
+def _compare_unknown_mo(from_mo, to_mo, diff):
+    """
+    Internal function to compare if any or both of the  ref and diff obj is
+    unknown mo.
+    """
+
+    # both unknown mo
+    for prop in from_mo.properties:
+        if prop not in to_mo.properties:
+            continue
+        if from_mo.properties[prop] != to_mo.properties[prop]:
+            diff.append(prop)
+
+
+def _compare(from_mo, to_mo, diff, include_operational=False,
+             no_version_filter=False):
+    """
+    Internal method to support compare reference and difference object.
+    """
+
+    # compare mo of different types
+    if from_mo.get_class_id() != to_mo.get_class_id():
+        return _CompareStatus.TYPES_DIFFERENT
+
+    # compare for unknown class_id
+    if isinstance(from_mo, GenericMo) or isinstance(to_mo, GenericMo):
```

Comment:
Minor nit.. For consistency sake, can you move the version check inside the _compare_unknown_MO method..